x509/src/pkix_extensions.rs

```diff
@@ -328,30 +328,32 @@ pub struct PolicyMapping {
 pub struct NameConstraints<'a> {
     /// permittedSubtrees       [0]     GeneralSubtrees OPTIONAL,
     //#[asn1(context_specific = "0", optional = "true", tag_mode = "IMPLICIT")]
-    permitted_subtrees: Option<GeneralSubtrees<'a>>,
+    pub permitted_subtrees: Option<GeneralSubtrees<'a>>,
 
     /// excludedSubtrees        [1]     GeneralSubtrees OPTIONAL }
     //#[asn1(context_specific = "1", optional = "true", tag_mode = "IMPLICIT")]
-    excluded_subtrees: Option<GeneralSubtrees<'a>>,
+    pub excluded_subtrees: Option<GeneralSubtrees<'a>>,
 }
 
 const PERMITTED_SUBTREES_TAG: TagNumber = TagNumber::new(0);
 const EXCLUDED_SUBTREES_TAG: TagNumber = TagNumber::new(1);
 
-impl<'a> ::der::Decodable<'a> for NameConstraints<'a> {
-    fn decode(decoder: &mut ::der::Decoder<'a>) -> ::der::Result<Self> {
-        decoder.sequence(|decoder| {
-            let permitted_subtrees =
-                ::der::asn1::ContextSpecific::decode_implicit(decoder, ::der::TagNumber::N0)?
-                    .map(|cs| cs.value);
-            let excluded_subtrees =
-                ::der::asn1::ContextSpecific::decode_implicit(decoder, ::der::TagNumber::N1)?
-                    .map(|cs| cs.value);
-            Ok(Self {
-                permitted_subtrees,
-                excluded_subtrees,
-            })
+// impl<'a> ::der::Decodable<'a> for NameConstraints<'a> {
+//     fn decode(decoder: &mut ::der::Decoder<'a>) -> ::der::Result<Self> {
+//         decoder.sequence(|decoder| {
+impl<'a> DecodeValue<'a> for NameConstraints<'a> {
+    fn decode_value(decoder: &mut Decoder<'a>, _length: Length) -> der::Result<Self> {
+        let permitted_subtrees =
+            ::der::asn1::ContextSpecific::decode_implicit(decoder, ::der::TagNumber::N0)?
+                .map(|cs| cs.value);
+        let excluded_subtrees =
+            ::der::asn1::ContextSpecific::decode_implicit(decoder, ::der::TagNumber::N1)?
+                .map(|cs| cs.value);
+        Ok(Self {
+            permitted_subtrees,
+            excluded_subtrees,
         })
+        // })
```

Comment:
Because it didn't work. We could revert and compare cargo expand to what I manually did. There have been some changes to derive since these edits were made so it is possible these manual edits are not needed.